docs/CONTRIBUTING_DEVELOPMENT.md

```diff
@@ -0,0 +1,262 @@
+component, please mention the issue containing the existing specification in
+your proposal. If the original issue hasn't been updated in a while, please ping
+@coryrylan, @gnomeontherun, or @mathisscott. We will start the discussion as
+soon as possible.
+
+We will discuss the proposal with you publicly on the issue, potentially
+requesting changes, and hopefully, accept it.
+
+## II. Implementation on a topic branch
+
+### Prerequisites
+
+Make sure you first:
+
+* Read our [Developer Certificate of Origin](https://cla.vmware.com/dco). All
+  contributions to this repository must be signed as described on that page.
+  Your signature certifies that you wrote the patch or have the right to pass it
+  on as an open-source patch.
+* Read our [Angular coding guidelines](/CODING_GUIDELINES_ANGULAR.md) and [Web Component coding guidelines](/CODING_GUIDELINES_CORE.md).
+
+### Getting started
+
+When we post on the issue to approve your proposal, the person on the team
+who'll be your primary contact will post a link to a topic branch against which
+you will submit your pull requests. The topic branch will be branched from the
+latest `master` and named `topic/{feature-name}`. To merge any pull request into
+this branch, it will need 2 approvals from team members.
+
+Start by [forking](https://help.github.com/articles/fork-a-repo/) the main
+Clarity repository, and follow the instructions in the previous link to clone
+your fork and set the upstream remote to the main Clarity repository. Because of
+the DCO, set your name and e-mail in the Git configuration for signing. Finally,
+create a local topic branch from the upstream `topic/{feature-name}` mentioned
+above.
+
+For instance, this setup part could look like this:
+
+```shell
+## Clone your forked repository
+git clone git@github.com:<github username>/clarity.git
+
+## Navigate to the directory
+cd clarity
+
+## Set name and e-mail configuration
+git config user.name "John Doe"
+git config user.email johndoe@example.com
+
+## Setup the upstream remote
+git remote add upstream https://github.com/vmware/clarity.git
+
+## Check out the upstream a topic branch for your changes
+git fetch
+git checkout -b topic/feature-name upstream/topic/feature-name
+```
+
+### Starting the project
+
+Once you have the project checked out with a fork you will need to run some step
+steps.
+
+1.  Have [NodeJS](https://nodejs.org) installed
+2.  In the root project directory run, `npm install`
+3.  Run `npm run build:ci` to build entire project (this may take several minutes)
+4.  Startup the project
+    * If Angular change run `npm run start`
+    * If Web Component change run `npm run core:start`
+    * If Website Documentation change run `npm run website:start`
+5.  To run tests and other project-specific commands see our project [Build Guide](/docs/BUILD.md)
+
+### Commits
+
+If your contribution is large, split it into smaller commits that are logically
```

Comment:
A big chunk of text that's very important. Let's break it up for the Twitter generation...

```
If your contribution is large, split it into smaller commits that are logically
self-contained. You can then submit them as separate pull requests that we will
review one by one and merge progressively into the topic branch. 

As a rule of thumb, try to keep each pull request under a couple of hundred lines of code,
_unit tests included_. We realize this isn't always easy, and sometimes not
possible at all, so feel free to ask how to split your contribution in the
GitHub issue. 

In general, it's a good idea to start coding the services first
and test them in isolation, then move to the components.
```